.github/workflows/codeql.yml

```diff
@@ -0,0 +1,82 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+  schedule:
+    - cron: '45 21 * * 6'
+
+jobs:
+  analyze:
+    name: Analyze
+    # Runner size impacts CodeQL analysis time. To learn more, please see:
+    #   - https://gh.io/recommended-hardware-resources-for-running-codeql
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners
+    # Consider using larger runners for possible analysis time improvements.
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
```

Comment:
Guessing this part came from a template / example; we don't have swift in this repository, which effectively means that CodeQL now only runs on `ubuntu-latest`.

I guess we can remove the `swift` condition, or make it explicitly only run on `ubuntu-latest`, or if running it on different platforms improves coverage of CodeQL, perhaps we should have all of Linux, macOS, and Windows

Reply:
Yeah, it's from the auto-generated stuff, I'll clean it up

Reply:
Sorry for being nit-picky; I expect these kind of changes to apply to other repositories where we'll be adding CodeQL as well, so probably after we tweaked things, we can copy/pasta things to the other repositories 🤗

Reply:
No worries at all!